cli/azd/pkg/project/framework_service_swa.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/swa"
+)
+
+type swaProject struct {
+	env           *environment.Environment
+	console       input.Console
+	commandRunner exec.CommandRunner
+	swa           swa.SwaCli
+	framework     FrameworkService
+}
+
+// NewSwaProject creates a new instance of a Azd project that
+// leverages swa cli for building
+func NewSwaProject(
+	env *environment.Environment,
+	console input.Console,
+	commandRunner exec.CommandRunner,
+	swa swa.SwaCli,
+	framework FrameworkService,
+) CompositeFrameworkService {
+	return &swaProject{
+		env:           env,
+		console:       console,
+		commandRunner: commandRunner,
+		swa:           swa,
+		framework:     framework,
+	}
+}
+
+// NewSwaProjectAsFrameworkService is the same as NewSwaProject().(FrameworkService) and exists to support our
+// use of DI and ServiceLocators, where we sometimes need to resolve this type as a FrameworkService instance instead
+// of a CompositeFrameworkService as [NewSwaProject] does.
+func NewSwaProjectAsFrameworkService(
+	env *environment.Environment,
+	console input.Console,
+	commandRunner exec.CommandRunner,
+	swa swa.SwaCli,
+	framework FrameworkService,
+) FrameworkService {
+	return NewSwaProject(env, console, commandRunner, swa, framework)
+}
+
+func (p *swaProject) Requirements() FrameworkRequirements {
+	return FrameworkRequirements{
+		Package: FrameworkPackageRequirements{
+			RequireRestore: false,
+			RequireBuild:   true,
+		},
+	}
+}
+
+// Gets the required external tools for the project
+func (p *swaProject) RequiredExternalTools(context.Context) []tools.ExternalTool {
+	return []tools.ExternalTool{p.swa}
```

Comment:
Should we take `p.inner.RequiredExternalTools` into account and then append this to it?  That way we ensure that the user has the underlying tool chain for the language?

Reply:
The `CompositeFrameworkService` gives us the impression that it uses the inner `FrameworkService`, however, that's not really true.
For Docker, for example, the folks can use a language not supported by azd, like go or ruby.
That applies to swa as well.

If the inner `FrameworkService` is really used by the `CompositeFrameworkService` , it would limit a `CompositeFrameworkService` to the universe of languages supported by azd